### PR TITLE
banana-vera: init at 1.3.0

### PR DESCRIPTION
**Pull request by Sigmanificient**
###### Description of changes

This package is used for analysis of C/C++ in order to create a linter easily.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by COLAMAroro — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/analysis/default.nix
Comment (Sigmanificient) [reply pkgs/development/tools/analysis/default.nix:1]: done

File: pkgs/development/tools/analysis/default.nix
Comment (Sigmanificient) [reply pkgs/development/tools/analysis/default.nix]: done

**Comment by COLAMAroro** [comment]
With requested change, the package builds & runs fine on `x86_64-linux` and `aarch64-linux`

File: pkgs/top-level/all-packages.nix
Comment (COLAMAroro) [inline pkgs/top-level/all-packages.nix:438]: I think there should be another opinion on this, but maybe consider renaming this to `banana-vera` ?

Epitech is the upstream maintainer, but the fork itself is called `banana-vera`

File: pkgs/development/tools/analysis/vera/default.nix
Comment (COLAMAroro) [inline pkgs/development/tools/analysis/vera/default.nix:14]: Consider using the [upstream bitbucket](https://bitbucket.org/verateam/vera/wiki/Home). The Github is just a replication

```suggestion
  src = fetchFromBitbucket {
```

**Review by COLAMAroro — CHANGES_REQUESTED** [review]
Current commits are tested through:
`NIXPKGS_ALLOW_INSECURE=1 nix run .#vera --impure`
`NIXPLGS_ALLOW_INSECURE=1 nix run .#legacyPackages.aarch64-linux.vera --impure`
`nix run .#vera-epitech`
`nix run .#legacyPackages.aarch64-linux.vera-epitech`

All builds and run fine.

**Review by natsukium — CHANGES_REQUESTED** [review]
Python2 has been EOL for over three years. What about not packaging upstream that is not updated and only adding the `banana-vera`?

**Comment by COLAMAroro** [comment]
The issue is, I'm not sure what the fork really does differently. Looking at the diff, it seems to only change Python and Boost version, but I'm not versed enough into integrating Python into an app, so I'm not sure.

If anyone with more experience can say for sure it's not doing any major changes into this, I guess it's fair to remove the upstream

https://github.com/verateam/vera/compare/master...Epitech:banana-vera:master

**Comment by Sigmanificient** [comment]
Ngl, seems like the fork is just a maintained version of vera

**Comment by natsukium** [comment]
OK, let's drop `vera`

**Comment by Sigmanificient** [comment]
> OK, let's drop `vera`

Done!

File: pkgs/development/tools/analysis/banana-vera/default.nix
Comment (natsukium) [inline pkgs/development/tools/analysis/banana-vera/default.nix:27]:
```suggestion
    python310.pkgs.boost
```
`boost` on python-packages already has python-enabled `include/`.

File: pkgs/development/tools/analysis/banana-vera/default.nix
Comment (natsukium) [inline pkgs/development/tools/analysis/banana-vera/default.nix:6]:
```suggestion
```

File: pkgs/development/tools/analysis/banana-vera/default.nix
Comment (natsukium) [inline pkgs/development/tools/analysis/banana-vera/default.nix:10]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```
We need to add this for `src.rev`.

File: pkgs/development/tools/analysis/banana-vera/default.nix
Comment (natsukium) [inline pkgs/development/tools/analysis/banana-vera/default.nix:44]:
```suggestion
})
```

File: pkgs/development/tools/analysis/banana-vera/default.nix
Comment (natsukium) [inline pkgs/development/tools/analysis/banana-vera/default.nix:17]:
```suggestion
    rev = "refs/tags/v${finalAttrs.version}";
```

File: pkgs/development/tools/analysis/banana-vera/default.nix
Comment (natsukium) [inline pkgs/development/tools/analysis/banana-vera/default.nix:18]:
```suggestion
    hash = "sha256-1nAKhUltQS1301JNrr0PQQrrf2W9Hj5gk1nbUhN4cXw=";
```

File: pkgs/development/tools/analysis/banana-vera/default.nix
Comment (natsukium) [inline pkgs/development/tools/analysis/banana-vera/default.nix:43]: Would you please add platforms?

File: pkgs/development/tools/analysis/banana-vera/default.nix
Comment (natsukium) [inline pkgs/development/tools/analysis/banana-vera/default.nix:37]:
```suggestion
    license = lib.licenses.boost;
```
LGTM except this.

**Review by natsukium — APPROVED** [review]

File: pkgs/development/tools/analysis/banana-vera/default.nix
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/banana-vera/default.nix:22]: Why is 310 hard coded here? Can we just use python3?

Comment (COLAMAroro) [reply pkgs/development/tools/analysis/banana-vera/default.nix:22]: I assume it's because upstream explicitly specified Python 3.10 [in the release](https://github.com/Epitech/banana-vera/releases/tag/v1.3.0-python3.10) and [in commit name](https://github.com/Epitech/banana-vera/commit/3a7d18d6249cd1790cc63fb9018e0914462ec219)

Comment (Sigmanificient) [reply pkgs/development/tools/analysis/banana-vera/default.nix:22]: Yes, the python version is pinned as python3.10

**Comment by SuperSandro2000** [comment]
@ofborg build banana-vera

**Pull request merged**
